### ISSUE TITLE
Materialize in Docker doesn't work with IPv6

### ISSUE THREAD

**Issue by antiguru** [MaterializeInc/materialize#12569]
### What version of Materialize are you using?

42b0545dd0e6fdc7ae7b893b845aeb773792b5ab

### How did you install Materialize?

Built from source

### What is the issue?

When running materialize through docker, Docker maps the port 6875 to two different ports on the host, one for IPv4 and one for IPv6. When connecting to the materialize instance, we don't specify which port should be used and might use either (not 100% sure about this). Materialize listens on the IPv4 port, and doesn't seem to allow successful connections on the IPv6 mapped port.

This could be an issue specific to Debian, but let's still take it as an action item that listening on `0.0.0.0` doesn't (or does!) imply listening on the corresponding IPv6 address.

TL;DR: I can connect to materialize using IPv6 on the IPv6 mapped port, and using IPv4 on the IPv4 mapped port, but not the other way around. The ssl problem is a red herring where `psql` tries to establish a connection using ssl if the plain connection fails.

docker ps:
```
CONTAINER ID   IMAGE                                                               COMMAND                  CREATED              STATUS              PORTS                                                                                      NAMES
8506d4aa70d4   materialize/materialized:mzbuild-CS2G7PBMCRNVI3IPO2UR6CCURS25TLBD   "tini -- entrypoint.…"   30 minutes ago       Up 30 minutes       0.0.0.0:49379->2101/tcp, :::49377->2101/tcp, 0.0.0.0:49377->6875/tcp, :::49376->6875/tcp   cluster_materialized_1
```

Different combinations of connecting to mz:
```
➜  materialize git:(main) ✗ psql -p 49377 -h 127.0.0.7 -U materialize
Timing is on.
psql (14.3 (Debian 14.3-1), server 9.5.0)
Type "help" for help.

materialize=>
\q
➜  materialize git:(main) ✗ psql -p 49377 -h localhost -U materialize
psql: error: connection to server at "localhost" (::1), port 49377 failed: received invalid response to SSL negotiation:
➜  materialize git:(main) ✗ psql -p 49377 -h ::1 -U materialize
psql: error: connection to server at "::1", port 49377 failed: received invalid response to SSL negotiation:
➜  materialize git:(main) ✗ psql -p 49377 -h ::1 -U materialize sslmode=disable
psql: error: connection to server at "::1", port 49377 failed: expected authentication request from server, but received
➜  materialize git:(main) ✗ psql -p 49377 -h localhost -U materialize sslmode=disable
psql: error: connection to server at "localhost" (::1), port 49377 failed: expected authentication request from server, but received
➜  materialize git:(main) ✗ psql -p 49377 -h 127.0.0.1 -U materialize sslmode=disable
Timing is on.
psql (14.3 (Debian 14.3-1), server 9.5.0)
Type "help" for help.

materialize=>
```

**Comment by antiguru** [comment]
Well, I couldn't make it work. What seems to work is disabling IPv6 within Docker :( I created a docker config as shown below:

`/etc/docker/docker.json` with the contents:
```json
{
        "ipv6": false
}
```

Now, `docker ps` shows that both the v4 and v6 ports are mapped to the same port on the host:
```
CONTAINER ID   IMAGE                                                               COMMAND                  CREATED         STATUS         PORTS                                                                                      NAMES
5098e1c73a40   materialize/materialized:mzbuild-SHDEHNDZNBUXLFE7TEV6TH4Y6Q76V6KZ   "tini -- entrypoint.…"   5 minutes ago   Up 5 minutes   0.0.0.0:49160->2101/tcp, :::49160->2101/tcp, 0.0.0.0:49159->6875/tcp, :::49159->6875/tcp   cluster_materialized_1
```
